tests/databricks/test_databricks.py

```diff
@@ -0,0 +1,213 @@
+    assert databricks_connector.get_status() == ConnectorStatus(
+        status=False,
+        message=None,
+        error='port closed',
+        details=[
+            ('Host resolved', True),
+            ('Port opened', False),
+            ('Connected to Databricks', None),
+            ('Authenticated', None),
+        ],
+    )
+    mocker.patch(
+        'pyodbc.connect',
+        side_effect=pyodbc.Error("I don't know mate"),
+    )
+    mocker.patch('toucan_connectors.databricks.databricks_connector.DatabricksConnector.check_port')
+    mocker.patch(
+        'toucan_connectors.databricks.databricks_connector.DatabricksConnector.check_hostname'
+    )
+    assert databricks_connector.get_status() == ConnectorStatus(
+        status=False,
+        message=None,
+        error="I don't know mate",
+        details=[
+            ('Host resolved', True),
+            ('Port opened', True),
+            ('Connected to Databricks', False),
+            ('Authenticated', None),
+        ],
+    )
+    mocker.patch(
+        'pyodbc.connect',
+        side_effect=pyodbc.InterfaceError(
+            'Invalid credentials', 'Authentication/authorization error occured'
+        ),
+    )
+    mocker.patch('toucan_connectors.databricks.databricks_connector.DatabricksConnector.check_port')
+    mocker.patch(
+        'toucan_connectors.databricks.databricks_connector.DatabricksConnector.check_hostname'
+    )
+    assert databricks_connector.get_status() == ConnectorStatus(
+        status=False,
+        message=None,
+        error='Invalid credentials',
+        details=[
+            ('Host resolved', True),
+            ('Port opened', True),
+            ('Connected to Databricks', True),
+            ('Authenticated', False),
+        ],
+    )
+    mocker.patch('pyodbc.connect')
+    mocker.patch('toucan_connectors.databricks.databricks_connector.DatabricksConnector.check_port')
+    mocker.patch(
+        'toucan_connectors.databricks.databricks_connector.DatabricksConnector.check_hostname'
+    )
+    assert databricks_connector.get_status() == CONNECTION_STATUS_OK
+
+
+@responses.activate
+def test_cluster_methods(databricks_connector: DatabricksConnector, mocker: MockFixture) -> None:
+    responses.add(
+        method='GET',
+        url='https://127.0.0.1/api/2.0/clusters/get',
+        headers={'login': 'token', 'password': '12345'},
+        match=[responses.matchers.json_params_matcher({'cluster_id': 'path'})],
+        json={'state': 'TERMINATED'},
+    )
+    assert not databricks_connector.check_cluster_running()
+    responses.add(
+        method='POST',
+        url='https://127.0.0.1/api/2.0/clusters/start',
+        headers={'login': 'token', 'password': '12345'},
+        match=[responses.matchers.json_params_matcher({'cluster_id': 'path'})],
+        json={},
+    )
+    sypied_start = mocker.spy(DatabricksConnector, 'start_cluster')
+    databricks_connector.start_cluster()
+    sypied_start.assert_called_once()
+
```

Comment:
I don't get the point of this ? We call the method and check that it has been called ?

Reply:
Or is it to ensure that it has not been called twice ?

Reply:
nope just a mistake. Ensuring that the correct call to URL is made is enough